docs/how-tos/protocol.rst

```diff
@@ -2,3 +2,536 @@
+
+1. **Settings class**: A Pydantic model defining configuration parameters
+2. **Protocol class**: The main class that creates computational workflows
+3. **ProtocolUnit classes**: Individual computational steps in the workflow
+4. **ProtocolResult class**: Container for aggregated results from multiple runs
+
+The ``Protocol`` creates a :ref:`ProtocolDAG <protocoldag>` (directed acyclic graph) of :ref:`ProtocolUnit <protocolunit>` objects that define the computational workflow.
+Multiple ``ProtocolDAG`` executions can be aggregated into a single :ref:`ProtocolResult <protocolresult>` to provide statistical estimates.
+
+
+Step 1: Define your Settings
+-----------------------------
+
+First, create a settings class that inherits from :class:`.Settings`.
+This defines all the configuration parameters your protocol needs:
+
+.. code-block:: python
+
+    from gufe.settings import Settings
+    import pydantic
+    from openff.units import unit
+
+    class MyProtocolSettings(Settings):
+        # Required: number of independent simulations
+        n_repeats: int = 5
+        
+        # Protocol-specific parameters
+        simulation_length: pydantic.Field(
+            default=10.0 * unit.nanosecond,
+            description="Length of each simulation"
+        )
+        
+        equilibration_length: pydantic.Field(
+            default=1.0 * unit.nanosecond, 
+            description="Equilibration time before production"
+        )
+
+
+Step 2: Define your ProtocolResult
+----------------------------------
+
+Create a result class that knows how to compute estimates and uncertainties from your protocol's outputs:
+
+.. code-block:: python
+
+    from gufe import ProtocolResult
+    from openff.units import unit
+    import numpy as np
+
+    class MyProtocolResult(ProtocolResult):
+        
+        def get_estimate(self) -> unit.Quantity:
+            """Calculate the free energy estimate from all runs."""
+            # Extract the key results from all completed runs
+            free_energies = []
+            for run_result in self.data["free_energies"]:
+                free_energies.append(run_result)
+            
+            # Return the mean as our best estimate
+            mean_dg = np.mean(free_energies) * unit.kilocalorie_per_mole
+            return mean_dg
+        
+        def get_uncertainty(self) -> unit.Quantity:
+            """Calculate the uncertainty from all runs."""
+            free_energies = self.data["free_energies"]
+            if len(free_energies) < 2:
+                return 0.0 * unit.kilocalorie_per_mole
+            
+            std_err = np.std(free_energies) / np.sqrt(len(free_energies))
+            return std_err * unit.kilocalorie_per_mole
+
+
+Step 3: Define your ProtocolUnits
+----------------------------------
+
+Create the computational units that will perform the actual work.
+Each unit should inherit from :class:`.ProtocolUnit` and implement a ``_execute`` method.
```

Comment:
```suggestion
Each unit should inherit from :ref:`.ProtocolUnit <protocolunit>` and implement an ``_execute`` method.
```

I think I prefer to link to the extensible points doc, since you can get from extensible points -> python api easily, but not vice-versa.